### ISSUE TITLE
Bundle analyzer `https://` imports displayed a bit confusingly 

### ISSUE THREAD

**Issue by marvinhagemeister** [evanw/esbuild#3423]
I'm currently playing around with `esbuild` in a Deno project and noticed that the bundle analyzer seems to parse `https://` specifiers as npm package names. 

<img width="1103" alt="Screenshot 2023-10-05 at 16 42 34" src="https://github.com/evanw/esbuild/assets/1062408/58a6a02e-ffff-407d-a7d3-fdaddb015340">

I think it would look much better if it would be able to parse the full origin of the URL and treat that as a folder. Something like:

- `https://esm.sh/`
- `v132/`
- ...

instead of:

- `https:/`
- `/`
- `esm.sh/`
- `v132/`
- ...

Don't know where the code for the visualizer is, otherwise I would've taken a stab at making a PR myself.

**Comment by hyrious** [comment]
It is in https://github.com/esbuild/esbuild.github.io/tree/main/src/analyze

**Comment by marvinhagemeister** [comment]
@hyrious awesome, thanks for sharing. Didn't know that there is a esbuild org on GitHub. Opened a PR over there to address this isse.

**Comment by evanw** [comment]
Can you provide a way to reproduce the issue? I'm not sure I understand what's going on here. Specific input files and build configuration with instructions would help.

For example, esbuild prefixes the path in the metafile with the namespace so when using the [example HTTP plugin](https://esbuild.github.io/plugins/#http-plugin) the prefix will be prefixed by `http-url:https://` instead, and your suggested fix won't work in that situation.

**Comment by marvinhagemeister** [comment]
@evanw Sure, here is the `metafile.json` that was generated: https://gist.github.com/marvinhagemeister/c2a0016856806ec9ae2dd48676308939 . It's from Fresh, a Deno project which uses a few https imports.

**Comment by evanw** [comment]
Ah I see, [their code](https://github.com/lucacasonato/esbuild_deno_loader/blob/570921fff647c27f589715bbc5954c9d1ceb81f7/src/shared.ts#L58) strips the protocol from the URL and passes it to esbuild as the namespace. Thanks that's helpful.

**Comment by marvinhagemeister** [comment]
Oh right. I didn't think of checking that. It's certainly something we can change. I'll take a look tomorrow why it was done this way.

**Comment by evanw** [comment]
No that's fine. That's a good use of the namespace feature. No need to change it.

I'm closing this issue because I have implemented this on https://esbuild.github.io/analyze/. My change seems to work well for your test case as well as the test case I have from the example HTTP plugin. Thanks for the test case. I also noticed that a data URL ended up in the bundle and the UI didn't work well because the data URL is really long, so I added some improvements there too (e.g. for tooltips).
